src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1328,13 +1325,21 @@ private LLVMValueRef LLVMFunctionForMethod(MethodDesc callee, StackEntry thisPoi
                     {
                         targetMethod = parameterType.ResolveInterfaceMethodTarget(callee);
                     }
+                    else if (constrainedType != null && constrainedType.IsSealed())
```

Comment:
It might be better to make this use `TryResolveConstraintMethodApprox` - it's is a copy of the equally named method in CLR.

While the logic you are adding here is similar to the current implementation of `TryResolveConstraintMethodApprox`, the actual correct logic is a bit more involved in the presence of shared code (we track fixing up `TryResolvedConstraintMethodApprox` in #2526).

Reply:
This code needs to work for reference types too, so I did a bit of refactoring to allow that. Please let me know if it looks reasonable.

Reply:
> This code needs to work for reference types too

Can you give an example? Why can't we just do an interface call? The code we ported from the CLR doesn't do anything for reference because `constrained` is pretty much only emitted in generic code and if the type operated on is a reference type, it's going to be `__Canon` at compile time and we can't resolve anything on it anyway.

Maybe right now it's resolving something in WASM, but that's just a temporary thing until shared generics are turned on. I don't think we'll ever need to support scenarios where shared generics are turned off (mostly because it would be really hard to find scenarios where the code bloat would justify the theoretical perf gain).

TryResolveConstraintMethodApprox is a weird method (and it's going to be even weirder with default interface methods), so I would really prefer if we could keep the structure equal with the structure on the CLR side.

Reply:
On further testing, we don't need that after all. I've removed those changes and just use this for value types